ranger-http/src/test/java/com/flipkart/ranger/http/serviceprovider/HttpShardedServiceProviderBuilderTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2015 Flipkart Internet Pvt. Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.ranger.http.serviceprovider;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.flipkart.ranger.core.healthcheck.Healthchecks;
+import com.flipkart.ranger.core.model.ServiceNode;
+import com.flipkart.ranger.http.config.HttpClientConfig;
+import com.flipkart.ranger.http.model.ServiceRegistrationResponse;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import lombok.Builder;
+import lombok.Data;
+import lombok.val;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+public class HttpShardedServiceProviderBuilderTest {
+
+    @Data
+    private static final class TestNodeData {
+        private final String farmId;
+
+        @Builder
+        public TestNodeData(@JsonProperty("farmId") String farmId) {
+            this.farmId = farmId;
+        }
+    }
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Rule
+    public WireMockRule server = new WireMockRule(8888);
+
+    @Test
+    public void testProvider() throws Exception {
+        val nm5NodeData = TestNodeData.builder().farmId("nm5").build();
```

Comment:
`nm5` -> some other string

Reply:
Oh great catch! Thank you. Completely missed it.